src/python/RucioUtils.py

```diff
@@ -64,26 +67,29 @@ def getWritePFN(rucioClient=None, siteName='', lfn='',
     # "write": provides the PFN to be used with gfal
     # 2022-08: dario checked with felipe that every sane RSE has non-zero value
     # for the third_party_copy_write column, which means that it is available.
-    ex_str = ""
+    exeptionString = ""
+    didDict = None
     for operation in operations:
         try:
             logger.warning('Try Rucio lfn2pn with operation %s', operation)
             didDict = rucioClient.lfns2pfns(siteName, [did], operation=operation)
             break
-        except Exception as ex:
+        except RSENotFound:
+            msg = f"Site {siteName} not found in CMS site list"
+            raise TaskWorkerException(msg) from RSENotFound
+        except Exception as ex:  # pylint: disable=broad-except
             msg = 'Rucio lfn2pfn resolution for %s failed with:\n%s\nTry next one.'
             logger.warning(msg, operation, str(ex))
-            ex_str += "operation: %s, exception: %s\n" % (operation, ex)
-            didDict = None
+            exeptionString += f"operation: {operation}, exception: {ex}\n"
     if not didDict:
-        msg = 'lfn2pfn resolution with Rucio failed for site: %s  LFN: %s' % (siteName, lfn)
-        msg += ' with exception(s) :\n%s' % str(ex_str)
+        msg = (f"lfn2pfn resolution with Rucio failed for site: {siteName}  LFN: {lfn}")
```

Comment:
Could you remove the parentheses from this line? 
It does not break, but to reduce confusion of the syntax.